src/lavinmq/auth/user.cr

```diff
@@ -130,9 +132,16 @@ module LavinMQ
         }
       end
 
-      def can_write?(vhost, name) : Bool
+      def can_write?(vhost : String, name : String, cache_key : CacheKey? = nil) : Bool
```

Comment:
There is no need for the caller to know that there is a cache (it maybe even isn't). Keep the old method signature and construct the cache key in `can_write?`. 
This will also simplify the entire method.